### PR TITLE
[Merged by Bors] - Refactor metrics

### PR DESCRIPTION
**Pull request by valar999**
## Motivation

Put metrics in order. Use one library for all metrics.
https://github.com/spacemeshos/pm/issues/37

## Changes

Refactor metrics in `sync`, `tortoise`, `hare` and `p2p` packages.

## TODO
- [x] Set metrics.Enable=true when Server or push-metrics are enabled 
- [ ] Decide is it good to use go-kit
- [ ] Find out what is MeteredCache is for

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by valar999** [comment]
A few comments about `go-kit`:
- it has no `Summary` and replace it with `Histogram`. But these are completely different things.
- go-kit allows to use variable in label names. We will have to remove this kind of variable https://github.com/spacemeshos/go-spacemesh/blob/v0.1.41/p2p/metrics/prometheus.go#L18 if we decide to get rid of go-kit

@noamnelke please consider

File: tortoise/metrics.go [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
+		[]string{"validity"},
+	)
+	invalidBlocks = blockVotes.With("validity", "invalid")
+
+	validBlocks = metrics.NewGauge(
+		"validity",
+		MetricsSubsystem,
+		"Valid",
+		nil,
+	)
```
Comment (noamnelke) [inline tortoise/metrics.go]: Why are these two so different?

**Review by noamnelke — APPROVED** [review]

File: tortoise/metrics.go (hunk 1 above)
Comment (valar999) [reply tortoise/metrics.go]: Thank you. It was a mistake. Fixed

**Comment by valar999** [comment]
bors merge

**Comment by valar999** [comment]
bors merge

**Comment by valar999** [comment]
bors merge

**Comment by valar999** [comment]
bors merge

**Pull request closed without merging**